src/Raven.Client/Http/RequestExecutor.cs

```diff
@@ -478,8 +484,13 @@ private void UpdateTopologyCallback(object _)
 
             _lastKnownUrls = initialUrls;
 
-            throw new AggregateException("Failed to retrieve cluster topology from all known nodes" + Environment.NewLine +
-                                         string.Join(Environment.NewLine, list.Select(x => x.Item1 + " -> " + x.Item2?.Message))
+            ThrowExceptions(list);
+        }
+
+        protected virtual void ThrowExceptions(List<(string, Exception)> list)
+        {
+            throw new AggregateException("Failed to retrieve database topology from all known nodes" + Environment.NewLine +
```

Comment:
If all exceptions are DatabaseDoesNotExistException then maybe it is better to throw that instead of AggregateException? Just an idea.

Reply:
We can't do that not all the exception will be DatabaseDoesNotExistException,
you can give the store a list of URLs when the you have more the one the requestExecutor will go and check on each of those URLs you can get different errors from different servers

Reply:
This is why I suggested to throw it when _all_ exceptions are DatbaseDoesNotExistException

Reply:
You will have to check all your error list to make sure every each of the exception are DatabaseDoesNotExistException